script/github-actions/select-cypress-tests.js

```diff
@@ -244,17 +252,63 @@ function exportVariables(tests) {
   core.exportVariable('TESTS', tests);
 }
 
-function run() {
-  const pathsOfChangedFiles = process.env.CHANGED_FILE_PATHS.split(' ');
+function main() {
   const graph = dedupeGraph(buildGraph());
-  const tests = selectTests(graph, pathsOfChangedFiles);
-  exportVariables(tests);
-}
 
-if (process.env.CHANGED_FILE_PATHS || RUN_FULL_SUITE) {
-  run();
+  // groups of tests from the allow list
+  const allAllowListTestPaths = ALLOW_LIST.map(spec => spec.spec_path);
+  const allAllowedTestPaths = ALLOW_LIST.filter(
+    spec => spec.allowed === true,
+  ).map(spec => spec.spec_path);
+  const allDisallowedTestPaths = ALLOW_LIST.filter(
+    spec => spec.allowed === false,
+  ).map(spec => spec.spec_path);
+
+  // groups of tests based on test selection and filtering the groups from the allow list
+  const testsSelectedByTestSelection = selectTests(graph, CHANGED_FILE_PATHS);
+  const newTests = testsSelectedByTestSelection.filter(
+    test =>
+      !allAllowListTestPaths.includes(test.substring(test.indexOf('src/'))),
+  );
+  const disallowedTests = testsSelectedByTestSelection.filter(test =>
+    allDisallowedTestPaths.includes(test.substring(test.indexOf('src/'))),
+  );
+  const changedTests = testsSelectedByTestSelection.filter(
+    test =>
+      CHANGED_FILE_PATHS.includes(test.substring(test.indexOf('src/'))) &&
+      !newTests.includes(test),
+  );
+  const testsToRunNormally = testsSelectedByTestSelection.filter(
+    test =>
+      !disallowedTests.includes(test) &&
+      !newTests.includes(test) &&
+      !changedTests.includes(test),
+  );
+  const testsToStressTest = [...newTests, ...changedTests];
+  const testSelectionDisallowedTests = testsSelectedByTestSelection.filter(
+    test => {
+      return allDisallowedTestPaths.includes(
+        test.substring(test.indexOf('src/')),
+      );
+    },
+  );
+
+  exportVariables(testsToRunNormally);
+
+  if (IS_STRESS_TEST) {
+    core.exportVariable('TESTS_TO_STRESS_TEST', allAllowedTestPaths);
+  } else {
+    core.exportVariable('TESTS_TO_STRESS_TEST', testsToStressTest);
+  }
+
+  core.exportVariable(
+    'TEST_SELECTION_DISALLOWED_TESTS',
+    testSelectionDisallowedTests,
+  );
 }
 
+main();
```

Comment:
This function call needs to be wrapped in an `if` statement since there are exports at the end of this file, otherwise the files importing from this file will run the `main` function.

We previously had this `if` statement to prevent this:
``` javascript
if (process.env.CHANGED_FILE_PATHS || RUN_FULL_SUITE) {
  run();
}
```

Reply:
Updated to:

```javascript
if (RUN_FULL_SUITE || ALLOW_LIST.length > 0) {
  main();
}
```

Reply:
It used to be:

```
if (process.env.CHANGED_FILE_PATHS || RUN_FULL_SUITE) {
  main();
}
```

Reply:
The updated conditional should work just fine.

Updated:
```javascript
if (RUN_FULL_SUITE || ALLOW_LIST.length > 0) {
  main();
}
```

From [`/.github/workflows/e2e-tests.yml`](https://github.com/department-of-veterans-affairs/vets-website/blob/main/.github/workflows/e2e-tests.yml#L150-L151)
```yml
      - name: Set TESTS variable
        run: node script/github-actions/select-cypress-tests.js
        env:
          RUN_FULL_SUITE: true
          CHANGED_FILE_PATHS: ''
```